lisa/_assets/kmodules/sched_tp/perf_counters.c

```diff
@@ -0,0 +1,694 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (C) 2023 ARM Ltd.
+#include <linux/perf_event.h>
+#if defined(CONFIG_HW_PERF_EVENTS) && defined(CONFIG_ARM_PMU)
+#include <linux/perf/arm_pmu.h>
+#endif
+#include "main.h"
+#include "features.h"
+#include "ftrace_events.h"
+#include "wq.h"
+
+#define MAX_PERF_COUNTERS	6
+#define HW_PERF_UNSUPPORTED	0xFFFFFFFF
+
+
+/* Initial set of supported counters to be enabled through module params */
+struct perfctr_desc {
+	const char 	*name;		/* unique name to identify the counter */
+	u64		id;		/* counter id (may be generic or raw) */
+	unsigned int	type;
+	unsigned int	default_on:1;	/* enable by default if no counters requested */
```

Comment:
I'd rather not have bitfields given the can of worms they are in general. You can simply use `bool` which is the appropriate type here.